### PR TITLE
feat(nova): implemented rating tutor functionality

### PR DESCRIPTION
**Pull request by mostafakamar2308**

### CODE REVIEW

**Comment by neuodev** [comment]
🚀 [Nova](https://nova-p5nm9gla1-litespace-core.vercel.app) 
🚀 [Luna](https://luna-nm2c1ajgd-litespace-core.vercel.app) 
🚀 [Nebula](https://nebula-heszdg8zx-litespace-core.vercel.app) 
🚀 [Dashboard](https://dashboard-7udqp4xmf-litespace-core.vercel.app)

**Comment by neuodev** [comment]
🚀 [Nova](https://nova-istuzint7-litespace-core.vercel.app) 
🚀 [Luna](https://luna-pyuena9d6-litespace-core.vercel.app) 
🚀 [Nebula](https://nebula-mxuyulnvv-litespace-core.vercel.app) 
🚀 [Dashboard](https://dashboard-hniombxdy-litespace-core.vercel.app)

**Comment by neuodev** [comment]
🚀 [Nova](https://nova-istuzint7-litespace-core.vercel.app) 
🚀 [Luna](https://luna-85ufibck4-litespace-core.vercel.app) 
🚀 [Nebula](https://nebula-n9j2xo258-litespace-core.vercel.app) 
🚀 [Dashboard](https://dashboard-f0gsakjb8-litespace-core.vercel.app)

File: apps/nova/src/components/TutorProfile/Ratings.tsx [hunk 1]
```diff
@@ -42,10 +50,72 @@ const Ratings: React.FC<{ id: number; tutorName: string | null }> = ({
...
+      toast.error({
+        title: intl("tutor.rating.delete.error"),
+      }),
+    [intl, toast]
+  );
+
+  const onDeleteSuccess = useCallback(() => {
+    toast.success({
+      title: intl("tutor.rating.delete.success"),
+    });
```
Comment (neuodev) [inline apps/nova/src/components/TutorProfile/Ratings.tsx:81]: No need to show a toast message. Student can see the result of his action already.

File: apps/nova/src/components/TutorProfile/Ratings.tsx [hunk 2]
```diff
@@ -42,10 +50,72 @@ const Ratings: React.FC<{ id: number; tutorName: string | null }> = ({
...
+  });
+
+  const closeEdit = useCallback(() => {
+    setEditDialog(null);
+  }, []);
+
+  const onEditSuccess = useCallback(() => {
+    toast.success({
+      title: intl("tutor.rating.edit.success"),
+    });
```
Comment (neuodev) [inline apps/nova/src/components/TutorProfile/Ratings.tsx:98]: Same here.

File: packages/luna/src/components/TutorFeedback/RatingDialog.tsx [hunk 3]
```diff
@@ -70,9 +77,12 @@ export const EditRating: React.FC<FeedbackEditProps> = ({
         <Button
           size={ButtonSize.Large}
           onClick={() =>
-            onUpdate({ value: newRating, feedback: newFeedback || null })
+            onSubmit({ value: newRating, feedback: newFeedback || "" })
```
Comment (neuodev) [inline packages/luna/src/components/TutorFeedback/RatingDialog.tsx:80]: Incase no feedback, it should be null. 


```suggestion
            onSubmit({ value: newRating, feedback: newFeedback || null })
```

File: packages/luna/src/components/TutorFeedback/TutorRatingCard.tsx [hunk 4]
```diff
@@ -78,7 +78,7 @@ export const TutorRatingCard: React.FC<RatingCardProps> = ({
             </div>
             <div
               className={cn(
-                "tw-bg-brand-500 tw-rounded-full tw-absolute tw-flex tw-justify-center tw-items-center tw-z-50",
+                "tw-bg-brand-500 tw-rounded-full tw-absolute tw-flex tw-justify-center tw-items-center tw-z-quote",
```
Comment (neuodev) [inline packages/luna/src/components/TutorFeedback/TutorRatingCard.tsx:81]: Nice!

File: packages/luna/src/components/TutorFeedback/TutorRatingCard.tsx [hunk 5]
```diff
@@ -98,8 +98,8 @@ export const TutorRatingCard: React.FC<RatingCardProps> = ({
         {!isEditing && !feedback ? null : (
           <Typography
             element="tiny-text"
-            weight={!feedback && isEditing ? "semibold" : "regular"}
-            className={cn("tw-line-clamp-5 tw-text-center tw-text-natural-600")}
+            weight={"semibold"}
```
Comment (neuodev) [inline packages/luna/src/components/TutorFeedback/TutorRatingCard.tsx:101]:
```suggestion
            weight="semibold"
```

File: apps/nova/src/components/TutorProfile/RateTutor.tsx [hunk 6]
```diff
@@ -0,0 +1,85 @@
...
+      toast.error({
+        title: intl("tutor.rate.error"),
+      }),
+    [intl, toast]
+  );
+
+  const onRateSuccess = useCallback(() => {
+    toast.success({
+      title: intl("tutor.rate.succes"),
+    });
```
Comment (neuodev) [inline apps/nova/src/components/TutorProfile/RateTutor.tsx:35]: No need to show a success toast.

**Review by neuodev — CHANGES_REQUESTED** [review]
Great work! I left minor comments. Please take a look. 


### UI/UX

The actions menu button on the card (the 3 dots) is not that usable, You need to be quite accurate when you click it. Let's increase the whitespace arount it such that even if the user didn't click it directly it should still open the actions menu. NOTE: this should note affect the design.

**Comment by neuodev** [comment]
🚀 [Nova](https://nova-a6v08r4uq-litespace-core.vercel.app) 
🚀 [Luna](https://luna-1co9llu0z-litespace-core.vercel.app) 
🚀 [Nebula](https://nebula-l5ux7jbmr-litespace-core.vercel.app) 
🚀 [Dashboard](https://dashboard-4qo0juo77-litespace-core.vercel.app)

**Comment by neuodev** [comment]
🚀 [Nova](https://nova-ltxq2daoj-litespace-core.vercel.app) 
🚀 [Luna](https://luna-2rhewwt43-litespace-core.vercel.app) 
🚀 [Nebula](https://nebula-ao5sw5onb-litespace-core.vercel.app) 
🚀 [Dashboard](https://dashboard-eqx2rhts5-litespace-core.vercel.app)

File: packages/luna/src/components/TutorFeedback/RatingDialog.tsx [hunk 7]
```diff
@@ -70,9 +77,12 @@ export const EditRating: React.FC<FeedbackEditProps> = ({
         <Button
           size={ButtonSize.Large}
           onClick={() =>
-            onUpdate({ value: newRating, feedback: newFeedback || null })
+            onSubmit({ value: newRating, feedback: newFeedback || null })
           }
-          disabled={newFeedback === feedback && newRating === rating}
+          disabled={
+            (newFeedback === feedback && newRating === rating) || loading
```
Comment (mmoehabb) [inline packages/luna/src/components/TutorFeedback/RatingDialog.tsx:83]: Just a simple note: consider using the short simple expression first in the (or) condition.

**Review by mmoehabb — COMMENTED** [review]
I've only left a simple humble note. What an incredible work @mostafakamar2308. Kudos to you.

**Review by mmoehabb — APPROVED** [review]

File: packages/luna/src/components/TutorFeedback/RatingDialog.tsx [hunk 8]
```diff
@@ -3,27 +3,32 @@ import { Dialog } from "@/components/Dialog/V2";
...
-import { FeedbackEditProps } from "@/components/TutorFeedback/types";
+import { RateDialogProps } from "@/components/TutorFeedback/types";
 import { Typography } from "@/components/Typography";
 import { useFormatMessage } from "@/hooks";
 import MessageQuestion from "@litespace/assets/MessageQuestion";
 import React, { useState } from "react";
 
-export const EditRating: React.FC<FeedbackEditProps> = ({
+/**
+ * This dialog will be used for both editing Ratings and creating new rating
```
Comment (neuodev) [inline packages/luna/src/components/TutorFeedback/RatingDialog.tsx:13]:
```suggestion
 * This dialog will be used for both editing ratings and creating new rating
```

**Review by neuodev — APPROVED** [review]

**Comment by neuodev** [comment]
@mostafakamar2308 please rebase, address the minor comment and merge you pull request. Thanks!

**Comment by neuodev** [comment]
🚀 [Nova](https://nova-hr7w9abuh-litespace-core.vercel.app) 
🚀 [Luna](https://luna-qq84rksio-litespace-core.vercel.app) 
🚀 [Nebula](https://nebula-5rzz0x0eo-litespace-core.vercel.app) 
🚀 [Dashboard](https://dashboard-nnp7e2ale-litespace-core.vercel.app)

**Pull request merged**